presto-hive/src/main/java/io/prestosql/plugin/hive/HiveTableHandle.java

```diff
@@ -29,28 +34,76 @@
 {
     private final String schemaName;
     private final String tableName;
-
+    private final List<ColumnHandle> partitionColumns;
+    private final Optional<List<HivePartition>> partitions;
+    private final TupleDomain<HiveColumnHandle> compactEffectivePredicate;
```

Comment:
In my understanding, `TupleDomain` need to be JSON serializable if we add it to `TableHandle`, but currently some classes which can be included in `TupleDomain` aren't JSON seraializable. Actually I'm facing this problem in #511. Making all classes which can be included in `TupleDomain` JSON serializable is a right way?

Reply:
`TupleDomain` should work for all types. `Domain` contains `ValueSet` and the two implementations with values `EquatableValueSet` and `SortedRangeSet` handle this by storing the value as a `Block` (which is always serializable in the engine via `BlockJsonSerde`).

Reply:
Oh, I see. Let me try to use `BlockJsonSerde`. Thanks!